samples/bluetooth/hci_rpmsg/src/main.c

```diff
@@ -0,0 +1,455 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <string.h>
+
+#include <zephyr.h>
+#include <arch/cpu.h>
+#include <sys/byteorder.h>
+#include <logging/log.h>
+#include <sys/util.h>
+#include <ipm.h>
+
+#include <openamp/open_amp.h>
+#include <metal/sys.h>
+#include <metal/device.h>
+#include <metal/alloc.h>
+
+#include <net/buf.h>
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/l2cap.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/buf.h>
+#include <bluetooth/hci_raw.h>
+
+#define LOG_LEVEL LOG_LEVEL_INFO
+#define LOG_MODULE_NAME hci_rpmsg
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+/* Configuration defines */
+
+#define SHM_START_ADDR      (0x20010000 + 0x400)
```

Comment:
Could you have  alook to this PR https://github.com/zephyrproject-rtos/zephyr/pull/17553 ?
An alternative is proposed to use  "chosen" declaration, in order to avoid this dependence on the platform in code.

Reply:
thanks for the tip, I cherry-picked commit to have ipm defines generated from dts